lib/cppcheck.cpp

```diff
@@ -229,13 +235,22 @@ static std::vector<std::string> split(const std::string &str, const std::string
     return ret;
 }
 
+static int getPid(void)
+{
+#ifndef _WIN32
+    return getpid();
+#else
+    return _getpid();
+#endif
+}
+
 static std::string getDumpFileName(const Settings& settings, const std::string& filename)
 {
     if (!settings.dumpFile.empty())
         return settings.dumpFile;
     if (!settings.dump && !settings.buildDir.empty())
-        return AnalyzerInformation::getAnalyzerInfoFile(settings.buildDir, filename, emptyString) + ".dump";
-    return filename + ".dump";
+        return AnalyzerInformation::getAnalyzerInfoFile(settings.buildDir, filename, emptyString) + "." + std::to_string(getPid()) + ".dump";
```

Comment:
Better move the extension into a separate variable to avoid duplicate calls.

Reply:
Done, thanks!